src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -0,0 +1,122 @@
+package io.iohk.ethereum.network.discovery
+
+import java.net.{InetSocketAddress, URI}
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import akka.agent.Agent
+import akka.util.ByteString
+import io.iohk.ethereum.rlp.RLPEncoder
+import io.iohk.ethereum.utils.{ByteUtils, NodeStatus, ServerStatus}
+import org.spongycastle.util.encoders.Hex
+
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class PeerDiscoveryManager(
+    discoveryListener: ActorRef,
+    discoveryConfig: DiscoveryConfig,
+    nodeStatusHolder: Agent[NodeStatus]) extends Actor with ActorLogging {
+
+  import PeerDiscoveryManager._
+
+  val nodesLimit = 200 // temporary limit until the whole protocol is implemented
+
+  var nodes: Map[ByteString, Node] = discoveryConfig.bootstrapNodes.map { s =>
+    val node = Node.parse(s)
+    node.id -> node
+  }.toMap
+
+  discoveryListener ! DiscoveryListener.Subscribe
+
+  context.system.scheduler.schedule(10.seconds, 1.minute) {
```

Comment:
maybe we can move to named val to know what it is
`scanStartDelay` `scanInterval`?